moonlight-core/src/main/scala/com/github/jamedge/moonlight/core/model/neo4j/queries/BaseQueriesConstructor.scala

```diff
@@ -0,0 +1,130 @@
+package com.github.jamedge.moonlight.core.model.neo4j.queries
+
+import com.github.jamedge.moonlight.core.model.neo4j.{GraphElement, Node, Nodes, RelationshipRight, Relationships}
+import neotypes.DeferredQueryBuilder
+import neotypes.implicits.all._
+
+class BaseQueriesConstructor[T <: Node](nodeFactory: String => T) {
+  def matchNode(name: String): DeferredQueryBuilder = {
+    matchNode(nodeFactory(name))
+  }
+
+  def matchNode(sourceNode: T): DeferredQueryBuilder = {
+    c"MATCH" + sourceNode.toSearchObject() + c"RETURN" + sourceNode.toVariable()
+  }
+
+  def matchDetails(sourceNodeName: String, lineName: String): DeferredQueryBuilder = {
+    matchDetails(nodeFactory(sourceNodeName), lineName)
+  }
+
+  def matchDetails(sourceNode: T, lineName: String): DeferredQueryBuilder = {
+    matchConnectingChain(sourceNode, List(ChainLink(DefaultRelationshipRightPresent(Relationships.HasDetails()), DefaultNodePresent(Nodes.Details(), unstructured = true))), lineName)
+  }
+
+  def matchConnectingNode(
+      sourceNodeName: String,
+      relationship: RelationshipRight,
+      destinationNode: Node,
+      lineName: String
+  ): DeferredQueryBuilder = {
+    matchConnectingChain(nodeFactory(sourceNodeName), List(ChainLink(DefaultRelationshipRightPresent(relationship), DefaultNodePresent(destinationNode))), lineName)
+  }
+
+  def matchConnectingChain(
+      sourceNodeName: String,
+      chain: List[ChainLink],
+      lineName: String
+  ): DeferredQueryBuilder = {
+    matchConnectingChain(nodeFactory(sourceNodeName), chain, lineName)
+  }
+
+  // TODO: split this into snippet functions
```

Comment:
yep, not to forget this. Additional structuring of this whole class is needed as well